cpp/ql/src/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -15,7 +15,11 @@ cached
 private newtype TIRDataFlowNode =
   TInstructionNode(Instruction i) or
   TOperandNode(Operand op) or
-  TVariableNode(Variable var)
+  TVariableNode(Variable var) or
+  // `FieldNodes` are used as targets of certain `storeStep`s to implement handling of stores to
+  // nested structs.
+  TFieldNode(FieldAddressInstruction field) or
+  TPartialDefinitionNode(PartialDefinition pd)
```

Comment:
With the addition of `TPartialDefinitionNode`, it looks to me like `TIRDataFlowNode` becomes recursive. Is that right? The `PartialDefinition` class is defined via a `Node node` member. I find this mildly confusing. Do you think it's necessary? Desirable?

Reply:
I agree that it's mildly confusing. The `Node node` member in the `PartialDefinition` class is never an instance of `PartialDefinition` so it's never a very deep recursion, though :) a00bd7ae027d17a2cd2e2931a37baa31943d1245 removed the `Node node` member so I guess it's slightly less confusing now.